core/steps/microvm/start_test.go

```diff
@@ -164,3 +170,42 @@ func TestNewStartStep_ServiceStartError(t *testing.T) {
 	g.Expect(subSteps).To(g.BeEmpty())
 	g.Expect(err).ToNot(g.BeNil())
 }
+
+func TestNewStartStep_unableToBoot(t *testing.T) {
+	g.RegisterTestingT(t)
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	microVMService := mock.NewMockMicroVMService(mockCtrl)
+	vm := testVMToStart()
+	ctx := context.Background()
+
+	step := microvm.NewStartStep(vm, microVMService, 1)
+
+	microVMService.
+		EXPECT().
+		Start(ctx, vm.ID.String()).
+		Return(nil)
+
+	microVMService.
+		EXPECT().
+		State(ctx, vm.ID.String()).
+		Return(ports.MicroVMStateUnknown, nil)
+
+	subSteps, err := step.Do(ctx)
+	verifyErr := step.Verify(ctx)
+
+	g.Expect(subSteps).To(g.BeEmpty())
+	g.Expect(err).To(g.BeNil())
+	g.Expect(verifyErr).To(g.MatchError(internalerr.ErrUnableToBoot))
```

Comment:
can we group these a little clearer?

```suggestion
	subSteps, err := step.Do(ctx)
	g.Expect(err).To(g.BeNil())
	g.Expect(subSteps).To(g.BeEmpty())
	
	verifyErr := step.Verify(ctx)
	g.Expect(verifyErr).To(g.MatchError(internalerr.ErrUnableToBoot))
```